features/bsm/service/impl/src/main/java/org/opennms/netmgt/bsm/service/internal/BusinessServiceManagerImpl.java

```diff
@@ -130,38 +131,22 @@ public boolean removeIpInterface(Long serviceId, Integer ipServiceId) {
     }
 
     @Override
-    public List<IpServiceDTO> getAllIpServiceDTO() {
-        return transformAll(monitoredServiceDao.findAll());
+    public OnmsSeverity getOperationalStatusForBusinessService(Long serviceId) {
+        final BusinessService service = getBusinessService(serviceId);
+        final OnmsSeverity severity = businessServiceStateMachine.getOperationalStatus(service);
+        return severity != null ? severity : OnmsSeverity.INDETERMINATE;
     }
 
     @Override
-    public OnmsSeverity calculateStatus(Long serviceId) {
-        final BusinessService service = getBusinessService(serviceId);
-        final List<BusinessStatusEntity> statusEntities = getDao().getStatus();
-        if (!statusEntities.isEmpty()) {
-            final Set<BusinessStatusEntity.Key> keySet = new HashSet<>();
-            for (OnmsMonitoredService eachMonitoredService : service.getIpServices()) {
-                BusinessStatusEntity.Key key = new BusinessStatusEntity.Key(eachMonitoredService.getNodeId(), eachMonitoredService.getServiceId(), eachMonitoredService.getIpAddress());
-                keySet.add(key);
-            }
+    public OnmsSeverity getOperationalStatusForIPService(Integer ipServiceId) {
+        final OnmsMonitoredService ipService = getIpService(ipServiceId);
+        final OnmsSeverity severity = businessServiceStateMachine.getOperationalStatus(ipService);
+        return severity != null ? severity : OnmsSeverity.INDETERMINATE;
```

Comment:
I guess, this should return OnmsSeverity.NORMAL instead of INDETERMINATE, otherwise the IT will fail (see below).